parsel/selector.py

```diff
@@ -4,16 +4,34 @@
 
 import typing
 import warnings
-from typing import Any, Dict, List, Mapping, Optional, Pattern, Union
+from typing import (
+    Any,
+    Dict,
+    List,
+    Mapping,
+    Optional,
+    Pattern,
+    Type,
+    TypeVar,
+    Union,
+)
 from warnings import warn
 
+from cssselect import GenericTranslator as OriginalGenericTranslator
 from lxml import etree, html
 from pkg_resources import parse_version
 
 from .csstranslator import GenericTranslator, HTMLTranslator
 from .utils import extract_regex, flatten, iflatten, shorten
 
-_SelectorType = typing.TypeVar("_SelectorType", bound="Selector")
+
+if typing.TYPE_CHECKING:
+    # requires Python 3.8
+    from typing import SupportsIndex
```

Comment:
I think it's fine to implicitly require that typing checks only work on more recent Python versions.